### PR TITLE
feat: Support lookup by trip instead of just vehicle

### LINKED ISSUE #11: Support looking up via trip instead of just vehicle
the '--trip' arg doesn't really work

### PR DESCRIPTION
**Pull request by greghutch**
Refactor log query logic slightly.  Still more to
be done once tasks are supported.

Fixes #11

### CODE REVIEW

**Comment by ewbao** [comment]
Looks like there are some link errors, but this LGTM.

**Review by bshi — APPROVED** [review]
nits

File: dune-buggy.js [hunk 1]
```diff
@@ -50,12 +50,41 @@ const yargs = require("yargs/yargs")(process.argv.slice(2))
   });
 const argv = yargs.argv;
 
+/*
+ * Extract trip_ids from vehicle logs, and query again to
```
[the remark this answers is not in the document]
Comment (greghutch) [reply dune-buggy.js:54]: fixed

File: dune-buggy.js [hunk 2]
```diff
@@ -50,12 +50,41 @@ const yargs = require("yargs/yargs")(process.argv.slice(2))
   });
 const argv = yargs.argv;
 
+/*
+ * Extract trip_ids from vehicle logs, and query again to 
+ */
+async function fetchTripLogsForVehicle(vehicleLogs, vehicle_id) {
+   console.log('Loading trip logs for vehicle', vehicle_id);
```
[the remark this answers is not in the document]
Comment (greghutch) [reply dune-buggy.js:57]: fixed

**Comment by greghutch** [comment]
> Looks like there are some link errors, but this LGTM.


Yeah.  I pushed a series of separate pull requests which should hopefully clear up those issues.

**Pull request merged**